macOS zig translate-c missing some bounds safety attributes

I'm unsure whether this issue belongs in arocc or in translate-c; if it's misplaced, please feel free to close it.

---

translate-c: https://codeberg.org/ziglang/translate-c

After updating to the latest arocc version 592f5b057865d29af163d495c6067b7ac444c109, when running translate-c tests on macOS or cross-compiling to macOS, some bounds safety attributes are missing, causing test failures.


### Reproduction

```bash
# git clone https://codeberg.org/ziglang/translate-c
cd tranlate-c
anyzig 0.16.0-dev.1484+d0ba6642b fetch --save git+https://github.com/Vexu/arocc#592f5b057865d29af163d495c6067b7ac444c109
anyzig 0.16.0-dev.1484+d0ba6642b build test -Doptimize=Debug -Dtest-cross-targets
```

Output:

```bash
home/happystraw/.cache/zig/p/N-V-__8AAMMAzhS44i3gROdtiKVCevpN_PbdgJBDAczz0KtB/lib/libc/include/any-darwin-any/malloc/_malloc_type.h:28:1: error: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]
__ptrcheck_abi_assume_single()
^
/home/happystraw/.cache/zig/p/N-V-__8AAMMAzhS44i3gROdtiKVCevpN_PbdgJBDAczz0KtB/lib/libc/include/any-darwin-any/malloc/_malloc_type.h:28:30: error: expected function body after function declaration
__ptrcheck_abi_assume_single()
                             ^
/home/happystraw/.cache/zig/p/N-V-__8AAMMAzhS44i3gROdtiKVCevpN_PbdgJBDAczz0KtB/lib/libc/include/any-darwin-any/malloc/_malloc_type.h:93:27: error: parameter 'size' was not declared, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]
void * __sized_by_or_null(size) malloc_type_malloc(size_t size, malloc_type_id_t type_id) __result_use_check __alloc_size(1);
                          ^
/home/happystraw/.cache/zig/p/N-V-__8AAMMAzhS44i3gROdtiKVCevpN_PbdgJBDAczz0KtB/lib/libc/include/any-darwin-any/malloc/_malloc_type.h:93:33: error: expected function body after function declaration
void * __sized_by_or_null(size) malloc_type_malloc(size_t size, malloc_type_id_t type_id) __result_use_check __alloc_size(1);
                                ^
/home/happystraw/.cache/zig/p/N-V-__8AAMMAzhS44i3gROdtiKVCevpN_PbdgJBDAczz0KtB/lib/libc/include/any-darwin-any/malloc/_malloc_type.h:93:33: error: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]
void * __sized_by_or_null(size) malloc_type_malloc(size_t size, malloc_type_id_t type_id) __result_use_check __alloc_size(1);
...
```

### Additional Information

Temporarily adding the missing macro definitions to `include/ptrcheck.h`, ran successful.

https://github.com/Vexu/arocc/blob/592f5b057865d29af163d495c6067b7ac444c109/include/ptrcheck.h#L9-L17 

```c
// ptrcheck.h
#define __counted_by(N)
#define __counted_by_or_null(N)
#define __sized_by(N)
#define __sized_by_or_null(N)
#define __ended_by(E)
#define __terminated_by(T)
#define __null_terminated
#define __ptrcheck_abi_assume_single()
```

--- a/include/ptrcheck.h
+++ b/include/ptrcheck.h
@@ -15,5 +15,35 @@
 
 #define __single
 #define __unsafe_indexable
+#define __counted_by(N)
+#define __counted_by_or_null(N)
+#define __sized_by(N)
+#define __sized_by_or_null(N)
+#define __ended_by(E)
+
+#define __terminated_by(T)
+#define __null_terminated
+
+/* __ptrcheck_abi_assume_indexable and __ptrcheck_abi_assume_bidi_indexable intentionally not defined */
+#define __ptrcheck_abi_assume_single()
+#define __ptrcheck_abi_assume_unsafe_indexable()
+
+#define __unsafe_forge_bidi_indexable(T, P, S) ((T)(P))
+#define __unsafe_forge_single(T, P) ((T)(P))
+#define __unsafe_forge_terminated_by(T, P, E) ((T)(P))
+#define __unsafe_forge_null_terminated(T, P) ((T)(P))
+
+#define __terminated_by_to_indexable(P) (P)
+#define __unsafe_terminated_by_to_indexable(P) (P)
+#define __null_terminated_to_indexable(P) (P)
+#define __unsafe_null_terminated_to_indexable(P) (P)
+#define __unsafe_terminated_by_from_indexable(T, P, ...) (P)
+#define __unsafe_null_terminated_from_indexable(P, ...) (P)
+
+#define __array_decay_dicards_count_in_parameters
+
+#define __ptrcheck_unavailable
+#define __ptrcheck_unavailable_r(REPLACEMENT)
+
 
 #endif

